schemas/2016-02-03/Microsoft.Devices.json

```diff
@@ -355,6 +370,53 @@
         }
       },
       "description": "The properties of the Storage Endpoint provided by the customer for file upload"
+    },
+    "IpFilterRule": {
+      "description": "IP filter Rule",
+      "type": "object",
+      "properties": {
+        "filterName": {
+          "oneOf": [
+            {
+              "description": "The name of the IP filter rule with the following regex pattern.",
+              "type": "string",
+              "pattern": "^[A-Za-z0-9-:.+%_#*?!(),=@;']{1,128}$",
+              "minLength": 1,
+              "maxLength": 128
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression" 
+            }
+          ]
+        },
+        "action": {
+          "oneOf": [
+            {
+              "description": "The action desired - accept or reject.",
+              "enum": [ "Accept", "Reject" ]
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression" 
+            }
+          ]
+        },
+        "ipMask": {
```

Comment:
"string" type property already support "expression" as expression itself is a string, the "oneOf" would make VS failed to match the property.
Please revert this ipMask property to last iteration.